include/tc/core/tc_executor.h

```diff
@@ -90,8 +97,26 @@ class TcExecutor {
         << "TcExecutor::uncheckedRun is abstract and should not be called";
   }
 
+  virtual bool hasRuntimeCompiledFun() {
+    LOG(FATAL)
+        << "TcExecutor::hasRuntimeCompiledFun is abstract and should not be called";
+    return true;
+  }
+
+  virtual void clearRuntimeCompiledFun() {
+    LOG(FATAL)
+        << "TcExecutor::clearRuntimeCompiledFun is abstract and should not be called";
+  }
+
   const static size_t InvalidHandle = std::numeric_limits<size_t>::max();
 
+  std::string identifier;
+  std::vector<dlutils::DLTensorUPtr> inputsInfo;
+  std::string options;
+  /// When run is called this is used to find the most recently compiled
+  /// version.
+  size_t objectLocalHandle;
```

Comment:
do we still need this thing?  it's redundant with the object's position in `tcExecutors_` list

Reply:
nice catch thanks!